Add bandit to supported linters

demus opened the pull request:
For #2726 

<!--
  If an item below does not apply to you, then go ahead and check it off as "done" and strikethrough the text, e.g.:
    - [x] ~Has unit tests & system/integration tests~
-->
- [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR)
- [x] Title summarizes what is changing
- [x ] Has a [news entry](https://github.com/Microsoft/vscode-python/tree/master/news) file (remember to thank yourself!)
- [x] Unit tests & system/integration tests are added/updated
- [x] ~Any new/changed dependencies in [`package.json`](https://github.com/Microsoft/vscode-python/blob/master/package.json) are pinned (e.g. `"1.2.3"`, not `"^1.2.3"` for the specified version)~
- [x] ~[`package-lock.json`](https://github.com/Microsoft/vscode-python/blob/master/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed)~

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Microsoft/vscode-python?pullRequest=2775) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Microsoft/vscode-python?pullRequest=2775) before we can accept your contribution.<br/><br/>:x: demus [sign now](https://cla.opensource.microsoft.com/Microsoft/vscode-python?pullRequest=2775)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Microsoft/vscode-python?pullRequest=2775) it.</sub>

demus commented:
The order of the linters in various data structures was kind of random. Should I alphabetize or standardize this?

DonJayamanne commented:
@brettcannon /cc

demus commented:
@DonJayamanne, thanks for the super fast review and feedback! will make those changes.

demus commented on src/client/common/configSettings.ts:
```diff
@@ -214,6 +215,11 @@ export class PythonSettings extends EventEmitter implements IPythonSettings {
+            banditCategorySeverity: {
+                LOW: DiagnosticSeverity.Error,
+                MEDIUM: DiagnosticSeverity.Error,
+                HIGH: DiagnosticSeverity.Error
```
whoops, fixed

demus commented on src/client/common/types.ts:
```diff
@@ -182,6 +183,11 @@ export interface IMypyCategorySeverity {
+    LOW: DiagnosticSeverity.Error;
+    MEDIUM: DiagnosticSeverity.Error;
+    HIGH: DiagnosticSeverity.Error;
+}
```
fixed

demus commented on package.json:
```diff
@@ -994,6 +994,55 @@
+                    ],
+                    "scope": "resource"
+                },
+                "python.linting.banditCategorySeverity.HIGH": {
```
done

demus commented on src/client/linters/bandit.ts:
```diff
@@ -0,0 +1,24 @@
+    protected async runLinter(document: TextDocument, cancellation: CancellationToken): Promise<ILintMessage[]> {
+        const messages = await this.run(['-f', 'custom', '--msg-template', '{line},0,{severity},{test_id}:{msg}', document.uri.fsPath], document, cancellation);
+        messages.forEach(msg => {
+            msg.severity = this.parseMessagesSeverity(msg.type, this.pythonSettings.linting.banditCategorySeverity);
```
done

demus commented:
@DonJayamanne Made all the changes. Added a fix as well for a bug in Bandit's message templating, documented here: https://github.com/PyCQA/bandit/issues/371 and fixed on Bandit master

demus commented on src/test/.vscode/settings.json:
```diff
@@ -18,5 +18,6 @@
     "python.formatting.provider": "yapf",
-    "python.linting.pylintUseMinimalCheckers": false
+    "python.linting.pylintUseMinimalCheckers": false,
+    "python.pythonPath": "python"
```
done.

DonJayamanne commented:
@demus 
Please could you resolve the code merge issue.
I'd like to merge this in today.

demus commented:
@DonJayamanne should be all set
